src/app/(builder)/builder/filteredTable.tsx

```diff
@@ -16,11 +16,18 @@ function includesIfFilter(filter: string | undefined, value: string) {
     return matchesIfFilter(filter, (f) => (value) ? value.toLowerCase().includes(f.toLowerCase()) : false);
 }
 
-function matchAbilities(filter: string, value: string) {
+function matchAbilities(filter: string, abilities: string) {
     const queries = filter.split(new RegExp("[, ]+"));
-    return queries.reduce((res, q) => {
-        return value?.toLowerCase().includes(q.toLowerCase().trim());
-    }, true);
+    const matches = queries.map((query) => {
+        query = query.toLowerCase().trim();
+        const neg = query.startsWith("!");
+        const contains = abilities
+          ?.toLowerCase()
+          .includes(neg ? query.slice(1) : query);
+        return neg ? !contains : contains;
+    }).every((match) => match === true);

```

Comment:
```suggestion
    const matches = queries.reduce((res, query) => {
        query = query.toLowerCase().trim();
        const cleanQ = query.replace(/^!/, '');
        const neg = q != cleanQ
        const contains = abilities?.toLowerCase().includes(cleanQ);
        return res && (neg ^ contains);
    }, true);
```
I think this is a bit cleaner this way? Also, appreciate the fix to the reduce bug - I forgot to _actually reduce_ the value. I knew the bug was there, but just didn't get around to fixing it.